frontend/src/randomPassword.js

```diff
@@ -1,52 +1,39 @@
-function getRandomInt(max) {
-    return Math.floor(Math.random() * max)
-}
-
-class PasswordCategoryPool {
-    characters;
-    constructor(characters) {
-        this.characters = characters;
-    }
-    genRandomChar() {
-        return this.characters.charAt(getRandomInt(this.characters.length));
-    }
-}
-
-function generatePassword(length, uppercase, lowercase, numbers, specialChars) {
-    const UppercasePasswordPool = new PasswordCategoryPool("ABCDEFGHIJKLMNOPQRSTUVWXYZ");
-    const LowercasePasswordPool = new PasswordCategoryPool("abcdefghijklmnopqrstuvwxyz");
-    const NumbersPasswordPool = new PasswordCategoryPool("0123456789");
-    const SpecialCharsPasswordPool = new PasswordCategoryPool("!@#$%^&*");
+export function generatePassword(length, uppercase, lowercase, numbers, specialChars) {
+    // Define character sets
+    const uppercaseChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    const lowercaseChars = "abcdefghijklmnopqrstuvwxyz";
+    const numberChars = "0123456789";
+    const specialCharChars = "!@#$%^&*";
 
     // Create a character pool based on options
-    let charPool = [];
-    if (uppercase) charPool.push(UppercasePasswordPool);
-    if (lowercase) charPool.push(LowercasePasswordPool);
-    if (numbers) charPool.push(NumbersPasswordPool);
-    if (specialChars) charPool.push(SpecialCharsPasswordPool);
+    let charPool = "";
+    if (uppercase) charPool += uppercaseChars;
+    if (lowercase) charPool += lowercaseChars;
+    if (numbers) charPool += numberChars;
+    if (specialChars) charPool += specialCharChars;
 
     // Check if any options are provided
-    if (charPool.length === 0) {
+    if (!charPool) {
         console.error("Please select at least one option for the password.");
         return "";
     }
 
     let password = "";
 
     // Ensure at least one character from each selected character set is included
-    if (uppercase) password += UppercasePasswordPool.genRandomChar();
-    if (lowercase) password += LowercasePasswordPool.genRandomChar();
-    if (numbers) password += NumbersPasswordPool.genRandomChar();
-    if (specialChars) password += SpecialCharsPasswordPool.genRandomChar();
+    if (uppercase) password += uppercaseChars.charAt(Math.floor(Math.random() * uppercaseChars.length));
+    if (lowercase) password += lowercaseChars.charAt(Math.floor(Math.random() * lowercaseChars.length));
+    if (numbers) password += numberChars.charAt(Math.floor(Math.random() * numberChars.length));
+    if (specialChars) password += specialCharChars.charAt(Math.floor(Math.random() * specialCharChars.length));
 
     // Generate the remaining characters
     for (let i = password.length; i < length; i++) {
-        let randomPoolIndex = getRandomInt(charPool.length);
-        password += charPool[randomPoolIndex].genRandomChar();
+        const randomIndex = Math.floor(Math.random() * charPool.length);
```

Comment:
@zhimw I notice you are using math.random() to construct passwords. math.random() is [not cryptographically secure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) so should not be used to generate passwords. MDN suggests using the [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues) for use cases like this.

Reply:
Thanks for the note! I replaced Math.Random() with Crypto: getRandomValues() method